### ISSUE TITLE
`spack checksum` finds wrong URLs

### ISSUE THREAD

**Issue by RasmusWL** [LLNL/spack#103]
I was having some problems getting spack to find the correct URL for files.

With these settings

```python
    homepage = "http://fishshell.com/"
    url      = "http://fishshell.com/files/2.2.0/fish-2.2.0.tar.gz"
    list_url = homepage
```

I get the following result (with wrong URLs):

```sh
$ spack checksum fish
==> Found 5 versions of fish.
  2.2.0     http://fishshell.com/fish-2.2.0.tar.gz
  2.1.2     http://fishshell.com/fish-2.1.2.tar.gz
  2.1.1     http://fishshell.com/fish-2.1.1.tar.gz
  2.1.0     http://fishshell.com/fish-2.1.0.tar.gz
  2.0.0     http://fishshell.com/fish-2.0.0.tar.gz

How many would you like to checksum? (default is 5, q to abort)
==> Downloading...
==> Trying to fetch from http://fishshell.com/fish-2.2.0.tar.gz

curl: (22) The requested URL returned error: 404 Not Found
==> Fetching from http://fishshell.com/fish-2.2.0.tar.gz failed.
==> Error: All fetchers failed for spack-stage-Slflbn
```

**Comment by adamjstewart** [comment]
I have (partially) resolved this issue with pull request #408. The reason why Spack couldn't download http://fishshell.com/fish-2.2.0.tar.gz is because that URL does not exist. By setting list_url and list_depth properly, Spack is now able to download any version of Fish that you want.

```bash
$ spack fetch fish@2.1.0
==> Warning: There is no checksum on file to fetch fish@2.1.0 safely.
  Fetch anyway? [y/N] y
==> Fetching with no checksum.
  Add a checksum or use --no-checksum to skip this check.
==> Trying to fetch from http://fishshell.com/files/2.1.0/fish-2.1.0.tar.gz
######################################################################## 100.0%
```

However, this does not solve the issue that Spack shouldn't find any remote versions at that location. With the fix in place, `spack versions` and `spack checksum` do not find any remote versions, although they should. But `spack install` and `spack fetch` now work, which I think is a lot more important.

Does anyone have any idea why Spack was previously finding other versions of Fish but at the wrong URL? Now it doesn't find any other versions of Fish but at the right URL. I have noticed that Spack has a lot of trouble finding versions of packages as soon as list_url and list_depth are necessary.

**Comment by adamjstewart** [comment]
Looks like this was fixed recently. I reverted the `list_url` back to its previous value in #2022. Once that is merged, you should be good to go.
